src/recipe/Baked.Recipe.Service.Application/CodeGeneration/CodeGenerationLayer.cs

```diff
@@ -1,46 +1,107 @@
 ﻿using Baked.Architecture;
 using Baked.Domain.Model;
-
+using System.Reflection;
+using System.Text;
 using static Baked.CodeGeneration.CodeGenerationLayer;
 
 namespace Baked.CodeGeneration;
 
-public class CodeGenerationLayer : LayerBase<GenerateCode>
+public class CodeGenerationLayer : LayerBase<GenerateCode, Compile>
 {
     readonly IGeneratedAssemblyCollection _generatedAssemblies = new GeneratedAssemblyCollection();
+    readonly IGeneratedFileCollection _generatedFiles = new GeneratedFileCollection();
+    readonly string _location = default!;
+
+    public CodeGenerationLayer()
+    {
+        _location = Path.GetDirectoryName(Assembly.GetEntryAssembly()?.Location) ??
+            throw new("'EntryAssembly' should have existed with valid location");
+        _location = Path.Combine(_location, Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development");
```

Comment:
setting this twice looks confusing. either use a local variable, or inline first statement